Source/Core/DiscIO/DirectoryBlob.cpp

```diff
@@ -1004,32 +1026,32 @@ u64 DirectoryBlobPartition::SetApploader(std::vector<u8> apploader, const std::s
 {
   bool success = false;
 
-  m_apploader = std::move(apploader);
-  if (m_apploader.size() < 0x20)
+  std::vector<u8> new_apploader = apploader;
+  if (new_apploader.size() < 0x20)
   {
     ERROR_LOG_FMT(DISCIO, "{} couldn't be accessed or is too small", log_path);
   }
   else
   {
-    const size_t apploader_size = 0x20 + Common::swap32(*(u32*)&m_apploader[0x14]) +
-                                  Common::swap32(*(u32*)&m_apploader[0x18]);
-    if (apploader_size != m_apploader.size())
+    const size_t apploader_size = 0x20 + Common::swap32(*(u32*)&new_apploader[0x14]) +
+                                  Common::swap32(*(u32*)&new_apploader[0x18]);
+    if (apploader_size != new_apploader.size())
       ERROR_LOG_FMT(DISCIO, "{} is the wrong size... Is it really an apploader?", log_path);
     else
       success = true;
   }
 
   if (!success)
   {
-    m_apploader.resize(0x20);
+    new_apploader.resize(0x20);
     // Make sure BS2 HLE doesn't try to run the apploader
-    Write32(static_cast<u32>(-1), 0x10, &m_apploader);
+    Write32(static_cast<u32>(-1), 0x10, &new_apploader);
   }
 
-  m_contents.AddReference(APPLOADER_ADDRESS, m_apploader);
+  m_contents.Add(APPLOADER_ADDRESS, new_apploader);
```

Comment:
If you stash away `new_apploader.size()` before this line, so that we don't have to use `new_apploader` on the line below, you could use `std::move` here. But I'll leave this one up to you.